crates/task-impls/src/quorum_vote.rs

```diff
@@ -0,0 +1,383 @@
+    /// For the `QuorumProposalRecv` event.
+    QuorumProposal,
+    /// For the `DACRecv` event.
+    Dac,
+    /// For the `VidDisperseRecv` event.
+    Vid,
+}
+
+/// Validate the quorum proposal.
+// TODO: Complete the dependency implementation.
+// <https://github.com/EspressoSystems/HotShot/issues/2710>
+#[allow(clippy::needless_pass_by_value)]
+fn validate_quorum_proposal<TYPES: NodeType>(
+    _quorum_proposal: Proposal<TYPES, QuorumProposal<TYPES>>,
+    _event_sender: Sender<Arc<HotShotEvent<TYPES>>>,
+) -> bool {
+    true
+}
+
+/// Validate the DAC.
+// TODO: Complete the dependency implementation.
+// <https://github.com/EspressoSystems/HotShot/issues/2710>
+#[allow(clippy::needless_pass_by_value)]
+fn validate_dac<TYPES: NodeType>(
+    _dac: DACertificate<TYPES>,
+    _event_sender: Sender<Arc<HotShotEvent<TYPES>>>,
+) -> bool {
+    true
+}
+
+/// Validate the VID share.
+// TODO: Complete the dependency implementation.
+// <https://github.com/EspressoSystems/HotShot/issues/2710>
+#[allow(clippy::needless_pass_by_value)]
+fn validate_vid<TYPES: NodeType>(
+    _disperse: Proposal<TYPES, VidDisperse<TYPES>>,
+    _event_sender: Sender<Arc<HotShotEvent<TYPES>>>,
+) -> bool {
+    true
+}
+
+/// Handler for the vote dependency.
+struct VoteDependencyHandle<TYPES: NodeType> {
+    /// View number to vote on.
+    view_number: TYPES::Time,
+    /// Event sender.
+    sender: Sender<Arc<HotShotEvent<TYPES>>>,
+}
+impl<TYPES: NodeType> HandleDepOutput for VoteDependencyHandle<TYPES> {
+    type Output = Vec<Arc<HotShotEvent<TYPES>>>;
+    async fn handle_dep_result(self, res: Self::Output) {
+        // Add this commitment check to test if the handler works, but this isn't the only thing
+        // that we'll need to check.
+        // TODO: Complete the dependency implementation.
+        // <https://github.com/EspressoSystems/HotShot/issues/2710>
+        let mut payload_commitment = None;
+        for event in res {
+            match event.as_ref() {
+                HotShotEvent::QuorumProposalValidated(proposal) => {
+                    let proposal_payload_comm = proposal.block_header.payload_commitment();
+                    if let Some(comm) = payload_commitment {
+                        if proposal_payload_comm != comm {
+                            return;
+                        }
+                    } else {
+                        payload_commitment = Some(proposal_payload_comm);
+                    }
+                }
+                HotShotEvent::DACValidated(cert) => {
+                    let cert_payload_comm = cert.get_data().payload_commit;
+                    if let Some(comm) = payload_commitment {
+                        if cert_payload_comm != comm {
+                            return;
+                        }
+                    } else {
+                        payload_commitment = Some(cert_payload_comm);
+                    }
+                }
+                _ => {}
```

Comment:
I think we also need to check that VID Share commitment matches the `payload_commitment`.  We don't do that in the current code either but I think we should

Reply:
Sounds good! I added a comment to line 87 without the implementation. This PR includes one commitment check just to test if the framework works, and I'll add more verifications when completing #2710.